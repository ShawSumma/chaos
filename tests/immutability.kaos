bool a1 = true
bool b1 = false
print a1
print b1
b1 = a1
print a1
print b1
a1 = false
print a1
print b1

num a2 = 5
num b2 = 3
print a2
print b2
b2 = a2
print a2
print b2
a2 = 1
print a2
print b2

num a3 = 3.5
num b3 = 7.1
print a3
print b3
b3 = a3
print a3
print b3
a3 = 1.2
print a3
print b3

str a4 = 'asdasdas'
str b4 = 'qweqweqwe'
print a4
print b4
b4 = a4
print a4
print b4
a4 = '12312'
print a4
print b4


bool a5 = true
bool b5 = a5
print a5
print b5
a5 = false
print a5
print b5

num a6 = 5
num b6 = a6
print a6
print b6
a6 = 1
print a6
print b6

num a7 = 3.2
num b7 = a7
print a7
print b7
a7 = 35.3
print a7
print b7

str a8 = 'asdasda'
str b8 = a8
print a8
print b8
a8 = 'qwqweqwe'
print a8
print b8


num x = 5
array y = [x, 2, 3]
print y
x = 32
print x
print y

list z = [1, 2, 3]
list t = z
print z
print t
z[0] = 5
print z
print t
