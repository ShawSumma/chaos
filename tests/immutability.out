true
false
true
true
false
true
5
3
5
5
1
5
3.500000
7.100000
3.500000
3.500000
1.200000
3.500000
asdasdas
qweqweqwe
asdasdas
asdasdas
12312
asdasdas
true
true
false
true
5
5
1
5
3.200000
3.200000
35.299999
3.200000
asdasda
asdasda
qwqweqwe
asdasda
[5, 2, 3]
32
[5, 2, 3]
[1, 2, 3]
[1, 2, 3]
[5, 2, 3]
[1, 2, 3]
{'a': 5, 'b': 2, 'c': 3}
32
{'a': 5, 'b': 2, 'c': 3}
